Nfield.Quota/QuotaFrameValidator.cs

```diff
@@ -16,44 +16,49 @@ public QuotaFrameValidator()
 
             RuleFor(qf => qf.VariableDefinitions)
                 .Must(HaveUniqueIds)
-                .WithMessage("Quota frame definitions contain a duplicate id. Duplicate id: '{DuplicateValue}'")
+                    .WithMessage("Quota frame definitions contain a duplicate id. Duplicate id: '{DuplicateValue}'")
+                    .WithErrorCode("duplicate-definition-id")
                 .Must(HaveUniqueVariableNames)
-                .WithMessage("Quota frame definitions contain a duplicate variable name. Duplicate name: '{DuplicateValue}'")
+                    .WithMessage("Quota frame definitions contain a duplicate variable name. Duplicate name: '{DuplicateValue}'")
+                    .WithErrorCode("duplicate-variable-name")
                 .Must(HaveUniqueLevelNamesPerVariable)
-                .WithMessage("Quota frame definitions contain a duplicate level name. Duplicate name: '{DuplicateValue}'")
+                    .WithMessage("Quota frame definitions contain a duplicate level name. Duplicate name: '{DuplicateValue}'")
+                    .WithErrorCode("duplicate-level-name")
                 .Must(HaveVariablesWithAtLeastOneLevel)
-                .WithMessage("Quota frame definitions has variables with no levels. Affected variable definition id: '{VariableDefinitionId}'")
+                    .WithMessage("Quota frame definitions has variables with no levels. Affected variable definition id: '{VariableDefinitionId}'")
+                    .WithErrorCode("no-levels")
                 .Must(HaveValidOdinVariableName)
-                .WithMessage(
-                    "Odin variable name invalid. Odin variable names can only contain numbers, letters and '_' and cannot be empty. They can only ​start with​ a letter. First character cannot be '_' or a number. Variable definition Id '{DefId}' with name '{DefName}' has an invalid Odin Variable Name '{InvalidOdin}'");
+                    .WithMessage("Odin variable name invalid. Odin variable names can only contain numbers, letters and '_' and cannot be empty. They can only ​start with​ a letter. First character cannot be '_' or a number. Variable definition Id '{DefId}' with name '{DefName}' has an invalid Odin Variable Name '{InvalidOdin}'")
+                    .WithErrorCode("invalid-odin-variable-name");
```

Comment:
It might be that .WithErrorCode() is ignored in response. 
Alternative is to use .WithState(), then it is definitely part of the response.

Reply:
OK let's revisit this when we've integrated in the API. Perhaps we can change the serialization there.